modules/light-clients/08-wasm/types/client_state_test.go

```diff
@@ -601,258 +602,100 @@ func (suite *TypesTestSuite) TestVerifyMembershipGrandpa() {
+					_ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					var payload types.SudoMsg
+					err := json.Unmarshal(sudoMsg, &payload)
+					suite.Require().NoError(err)
 
-// 				clientState := testingpath.EndpointA.GetClientState()
-// 				proof, proofHeight = suite.chainB.QueryProofForStore(transfertypes.StoreKey, key, int64(clientState.GetLatestHeight().GetRevisionHeight()))
+					suite.Require().NotNil(payload.VerifyMembership)
+					suite.Require().Equal(proofHeight, payload.VerifyMembership.Height)
+					suite.Require().Equal(path, payload.VerifyMembership.Path)
+					suite.Require().Equal(proof, payload.VerifyMembership.Proof)
+					suite.Require().Equal(value, payload.VerifyMembership.Value)
 
-// 				value = []byte(GetSimApp(suite.chainB).TransferKeeper.GetPort(suite.chainB.GetContext()))
-// 				suite.Require().NoError(err)
-// 			},
-// 			true,
-// 		},
-// 		{
-// 			"delay time period has passed", func() {
-// 				delayTimePeriod = uint64(time.Second.Nanoseconds())
-// 			},
-// 			true,
-// 		},
-// 		{
-// 			"delay time period has not passed", func() {
-// 				delayTimePeriod = uint64(time.Hour.Nanoseconds())
-// 			},
-// 			false,
-// 		},
-// 		{
-// 			"delay block period has passed", func() {
-// 				delayBlockPeriod = 1
-// 			},
-// 			true,
-// 		},
-// 		{
-// 			"delay block period has not passed", func() {
-// 				delayBlockPeriod = 1000
-// 			},
-// 			false,
-// 		},
-// 		{
-// 			"latest client height < height", func() {
-// 				proofHeight = testingpath.EndpointA.GetClientState().GetLatestHeight().Increment()
-// 			}, false,
-// 		},
-// 		{
-// 			"invalid path type",
-// 			func() {
-// 				path = ibcmock.KeyPath{}
-// 			},
-// 			false,
-// 		},
-// 		{
-// 			"failed to unmarshal merkle proof", func() {
-// 				proof = []byte("invalid proof")
-// 			}, false,
-// 		},
-// 		{
-// 			"consensus state not found", func() {
-// 				proofHeight = clienttypes.ZeroHeight()
-// 			}, false,
-// 		},
-// 		{
-// 			"proof verification failed", func() {
-// 				// change the value being proved
-// 				value = []byte("invalid value")
-// 			}, false,
-// 		},
-// 		{
-// 			"proof is empty", func() {
-// 				// change the inserted proof
-// 				proof = []byte{}
-// 			}, false,
-// 		},
-// 	}
-
-// 	for _, tc := range testCases {
-// 		suite.Run(tc.name, func() {
-// 			suite.SetupWasmTendermint() // reset
-// 			testingpath = ibctesting.NewPath(suite.chainA, suite.chainB)
-// 			testingpath.SetChannelOrdered()
-// 			suite.coordinator.Setup(testingpath)
+					bz, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
 
-// 			// reset time and block delays to 0, malleate may change to a specific non-zero value.
-// 			delayTimePeriod = 0
-// 			delayBlockPeriod = 0
+					return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+		},
+		{
+			"wasm vm returns invalid proof error",
```

Comment:
```suggestion
			"failure: wasm vm returns invalid proof error",
```
Nit: some of the other PRs are using `success` and `failure` in test case names. success means the callbackFn doesn't panic or error